.azure/azure-pipelines.ci.yml

```diff
@@ -188,6 +188,37 @@ stages:
       arch: x64
       tls: openssl
 
+#
+# Performance Tests
+#
+
+- stage: performance
+  displayName: Performance Testing
+  dependsOn:
+  - build_windows
+  - build_linux
+  jobs:
+  - template: ./templates/run-performance.yml
+    parameters:
+      image: windows-latest
+      platform: windows
+      arch: x64
+      tls: stub
+  - template: ./templates/run-performance.yml
+    parameters:
+      image: ubuntu-latest
+      platform: linux
+      arch: x64
+      tls: openssl
+
+- stage: performance_publish
+  displayName: Performance Results Publish
+  dependsOn:
+  - performance
+  jobs:
+  - template: ./templates/publish-performance.yml
```

Comment:
Does this really need to be a separate stage? What about making it a job of the stage above, but make it depend on all the other jobs?

Reply:
I can check if its possible. Not fully sure, because we actually name the jobs in the template script based off of parameters. It might work, but it would be ugly, and we'd have to manually add every new platform to the dep list. Whereas currently the dependency is really easy.

Reply:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml#use-the-output-variable-from-a-job-in-a-condition-in-a-subsequent-job

Reply:
That actually doesn't help, because we'd still be explicitly naming things. In theory I should just be able to depend on the job name. I just think thats harder to maintain.

Reply:
I pushed it just to see what it looks like and to double check that it does work.

Reply:
Ok. It does work, but I have to put the depends on in `publish-performance.yml`. Because its a template, I can't put it in the root script.